fiducial_vlam/src/vloc_node.cpp

```diff
@@ -158,11 +158,21 @@ namespace fiducial_vlam
             camera_pose_pub_->publish(t_map_camera_msg);
           }
 
-          // Publish odometry of the camera
+          // Publish odometry of the camera and/or the base.
+          // The camera_frame_id parameter is non-empty to publish the camera odometry.
+          // The base_frame_id parameter is non-empty to publish the base odometry.
           if (cxt_.publish_odom_) {
-            auto odom_msg = to_odom_message(stamp, t_map_camera);
-            add_fixed_covariance(odom_msg.pose);
-            camera_odometry_pub_->publish(odom_msg);
+            if (cxt_.camera_frame_id_.size()) {
```

Comment:
Both odom messages published to the same topic. They should be published to different topics.

Both odom messages use the camera frame id. The base odom should use the base frame id.

I need to provide both frame ids (since they vary from drone to drone in the swarm), but I don't need both odom messages published. Perhaps it's better to split ctx_.publish_odom_ into ctx_.publish_base_odom_ and ctx_.publish_camera_odom_.

Reply:
I think I have addressed your comments. 

1) different odom topics
2) correct child_frame_id for odom messages
3) flags to publish odom messages.

Thanks!

Reply:
Great, thx.